python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -649,3 +653,18 @@ def update_usages(self):
             # We don't limit GPU resources, as not all operators
             # use GPU resources.
             self._op_budgets[op].gpu = float("inf")
+
+        from ray.data._internal.execution.operators.base_physical_operator import (
+            AllToAllOperator,
+        )
+
+        # An all-to-all operator waits for all its input operator’s outputs before
+        # processing data. This often forces the input operator to exceed its object
+        # store memory budget. To prevent deadlock, we disable object store memory
+        # backpressure for the input operator.
+        for op in eligible_ops:
+            if any(
+                isinstance(next_op, AllToAllOperator)
+                for next_op in op.output_dependencies
+            ):
+                self._op_budgets[op].object_store_memory = float("inf")
```

Comment:
Not sure whether it's better to special-case `AllToAllOperator` here or add yet another method to the `PhysicalOperator` interface to handle this. Thinking special-casing might be better than adding a method that would only be overridden by one implementation

Initially, I was thinking we could use `throttling_disabled`, but we currently assume that all operators with `throttling_disabled` are metadata operators, so we can't use it for `MapOperator`